cmd/src/repos_add_kvp.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/src-cli/internal/api"
+)
+
+func init() {
+	usage := `
+Examples:
+
+  Add a key-value pair to a repository:
+
+    	$ src repos add-kvp -repo=repoID -key=mykey -value=myvalue
+
+  Omitting -value will create a tag (a key with a null value).
+`
+
+	flagSet := flag.NewFlagSet("add-kvp", flag.ExitOnError)
+	usageFunc := func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src repos %s':\n", flagSet.Name())
+		flagSet.PrintDefaults()
+		fmt.Println(usage)
+	}
+	var (
+		repoFlag  = flagSet.String("repo", "", `The ID of the repo to add the key-value pair to (required)`)
+		keyFlag   = flagSet.String("key", "", `The name of the key to add (required)`)
+		valueFlag = flagSet.String("value", "", `The value associated with the key. Defaults to null.`)
+		apiFlags  = api.NewFlags(flagSet)
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+		if *repoFlag == "" {
+			return errors.New("error: repo is required")
+		}
+
+		keyFlag = nil
+		valueFlag = nil
+		flagSet.Visit(func(f *flag.Flag) {
+			if f.Name == "key" {
+				key := f.Value.String()
+				keyFlag = &key
+			}
+
+			if f.Name == "value" {
+				value := f.Value.String()
+				valueFlag = &value
+			}
+		})
+		if keyFlag == nil {
+			return errors.New("error: key is required")
+		}
+
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+
+		query := `mutation addKVP(
+  $repo: ID!,
+  $key: String!,
+  $value: String,
+) {
+  addRepoKeyValuePair(
+    repo: $repo,
+    key: $key,
+    value: $value,
+  ) {
+    alwaysNil
```

Comment:
didn't you add errors [here](https://github.com/sourcegraph/sourcegraph/pull/45023#pullrequestreview-1201722757)?

Reply:
Yeah, the errors do get surfaced to the user:
```
% go run ./cmd/src repos add-kvp -repo=fakerepo -key=mykey
GraphQL errors: {
  "message": "invalid graphql.ID",
  "path": [
    "addRepoKeyValuePair"
  ]
}
exit status 1
```
But `alwaysNil` needs to be here because we need to specify a selection of fields from the return type of the GraphQL endpoint (which is `EmptyResponse` which only has one field, `alwaysNil`)

<img width="371" alt="Screen Shot 2022-12-05 at 4 11 02 PM" src="https://user-images.githubusercontent.com/70350613/205753396-3e360ee1-10df-47eb-8419-9ddde6dfea41.png">

If I remove the `alwaysNil` selection from the query string I get:
```
% go run ./cmd/src repos add-kvp -repo=fakerepo -key=mykey
GraphQL errors: {
  "locations": [
    {
      "column": 3,
      "line": 6
    }
  ],
  "message": "Field \"addRepoKeyValuePair\" of type \"EmptyResponse!\" must have a selection of subfields. Did you mean \"addRepoKeyValuePair { ... }\"?"
}
exit status 1
```

Reply:
Oh duh. Forgot that `alwaysNil` could still have an error returned